sql/core/src/main/scala/org/apache/spark/sql/jdbc/MsSqlServerDialect.scala

```diff
@@ -167,10 +167,15 @@ private object MsSqlServerDialect extends JdbcDialect {
     throw QueryExecutionErrors.commentOnTableUnsupportedError()
   }
 
+  // SQL Server does not support, it uses `getTopExpression` instead.
   override def getLimitClause(limit: Integer): String = {
     ""
   }
 
+  override def getTopExpression(limit: Integer): String = {
+    if (limit > 0) s"TOP ($limit)" else ""
```

Comment:
Just curious, MsSQL syntax doesn't allow `TOP 0`?

Reply:
TOP 0 is supported but 0 is used as a sentinel value to disable the pushdown, it is similar to how LIMIT works currently. I tested `df.select("a").limit(0)` query and it is optimised to ignore the query (?).